include/ygm/container/oversized_bag.hpp

```diff
@@ -0,0 +1,335 @@
+
+  /** @todo ----------------------------------------------unchanged----------------------------------------------*/
+  oversized_bag(const self_type &other)  // If I remove const it compiles
+      : m_comm(other.comm()), pthis(this), partitioner(other.comm()) {
+    pthis.check(m_comm);
+  }
+
+  oversized_bag(self_type &&other) noexcept
+      : m_comm(other.comm()),
+        pthis(this),
+        partitioner(other.comm()),
+        m_local_bag(std::move(other.m_local_bag)) {
+    pthis.check(m_comm);
+  }
+
+  oversized_bag &operator=(const self_type &other) { return *this = oversized_bag(other); }
+  /** @todo --------------------------------------------end unchanged--------------------------------------------*/
+
+  /** @todo This might be hard to reimplement, we'll need to look at if we can swap on fstream object or if we'll have to swap data
+   * then have other.open() and this->open() to reopen the insert file
+   */
+  oversized_bag &operator=(self_type &&other) noexcept {
+    /* ---This is the original code---
+    std::swap(m_local_bag, other.m_local_bag);
+    */
+    return *this;
+  }
+
+  using detail::base_async_insert_value<oversized_bag<Item>, for_all_args>::async_insert;
+
+  void async_insert(const Item &value, int dest) {
+    auto inserter = [](auto pcont, const value_type &item) {
+      pcont->local_insert(item);
+    };
+
+    m_comm.async(dest, inserter, this->get_ygm_ptr(), value);
+  }
+
+  /** @todo ----------------------------------------------unchanged----------------------------------------------*/
+  void async_insert(const std::vector<Item> &values, int dest) {
+    auto inserter = [](auto pcont, const std::vector<Item> &values) {
+      for (const auto &v : values) {
+        pcont->local_insert(v);
+      }
+    };
+
+    m_comm.async(dest, inserter, this->get_ygm_ptr(), values);
+  }
+  /** @todo --------------------------------------------end unchanged--------------------------------------------*/
+
+  /** @todo */
+  void local_insert(const Item &val) { /*---This is the original code--- m_local_bag.push_back(val); */ }
+
+  /** @todo */
+  void local_clear() { /*---This is the original code--- m_local_bag.clear(); */ }
+
+  /** @todo */
+  size_t local_size() const { /*---This is the original code--- return m_local_bag.size(); */ }
+
+  /** @todo */
+  size_t local_count(const value_type &val) const {
+    /*---This is the original code--- 
+    return std::count(m_local_bag.begin(), m_local_bag.end(), val);
+    */
+  }
+
+  /** @todo */
+  template <typename Function>
+  void local_for_all(Function fn) {
+    /*---This is the original code---
+    std::for_each(m_local_bag.begin(), m_local_bag.end(), fn);
+    */
+  }
+
+  /** @todo */
+  template <typename Function>
+  void local_for_all(Function fn) const {
+    /*---This is the original code---
+    std::for_each(m_local_bag.cbegin(), m_local_bag.cend(), fn);
+    */
+  }
+  
+  /** @todo */  
+  void serialize(const std::string &fname) {
```

Comment:
Unsure of exactly what is expected from the serialization/deserialization process. I see cereal is utilized and a JSON Output is created, is it essentially putting the contents of a bag into a JSON format?

Reply:
These are for the ygm::bag, I'm not sure how much sense it makes to write the collective file contents to a JSON, I'll ask about it but we should be able to pull data from files and place into the json archive to support it.